### PR TITLE
feat: add new Perl pipelines for Makefile generation and cleanup

### PR DESCRIPTION
**Pull request by stormqueen1990**
Add two new Perl pipelines for repetitive tasks: Makefile generation based on a Perl module and cleanup after installation.

### CODE REVIEW

File: pkg/build/pipelines/perl/cpan-install.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: Install a Perl package downloaded from cpan
```
Comment (vaikas) [inline pkg/build/pipelines/perl/cpan-install.yaml:1]: Nit, but the way I read the pipeline, I understood that it just creates the Makefile that you then will have to use in `make` later

**Review by vaikas — COMMENTED** [review]
Awesomesauce!!!
Just curious about the install vs. construct Makefile and it's probably just my misunderstanding 🤣

File: pkg/build/pipelines/perl/cpan-install.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+needs:
+  packages:
+    - busybox
+    - perl
+
+pipeline:
+  - runs: |
+      export CFLAGS=$(perl -MConfig -E 'say $Config{ccflags}')
```
[the remark this answers is not in the document]
Comment (kaniini) [reply pkg/build/pipelines/perl/cpan-install.yaml:10]: I think we should capture that output into $PERL_CFLAGS and then append it to $CFLAGS.

File: pkg/build/pipelines/perl/cpan-install.yaml (hunk 1 above)
Comment (stormqueen1990) [reply pkg/build/pipelines/perl/cpan-install.yaml:1]: That's true! I'll rename.

File: pkg/build/pipelines/perl/cpan-install.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (stormqueen1990) [reply pkg/build/pipelines/perl/cpan-install.yaml]: Changed it to append. Is space OK between contents?

**Review by vaikas — APPROVED** [review]

**Review by kaniini — APPROVED** [review]

**Pull request merged**